### ISSUE TITLE
Return `factor` in SpeedPerturbation

### ISSUE THREAD

**Issue by cassiotbatista** [pytorch/audio#3626]
### 🚀 The feature

[`SpeedPerturbation`](https://github.com/pytorch/audio/blob/main/torchaudio/transforms/_transforms.py#L1999) forward method should return the factor by which the signal was compressed or stretched or kept unaltered. The following snippet illustrates the feature, returning a nested tuple `((x, y), z)`.

```python
class SpeedPerturbation(torch.nn.Module):

    ...

    def __init__(self, orig_freq: int, factors: Sequence[float]) -> None:
        super().__init__()

        self.speeders = torch.nn.ModuleList([Speed(orig_freq=orig_freq, factor=factor) for factor in factors])
        self.factors = torch.nn.ModuleList(factors)  # XXX <- change here

    def forward(
        self, waveform: torch.Tensor, lengths: Optional[torch.Tensor] = None
    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], float]:

        ...

        idx = int(torch.randint(len(self.speeders), ()))
        # NOTE: we do this because TorchScript doesn't allow for
        # indexing ModuleList instances with non-literals.
        for speeder_idx, (speeder, factor) in enumerate(zip(self.speeders, self.factors)):
            if idx == speeder_idx:
                return speeder(waveform, lengths), factor  # XXX <- change here
        raise RuntimeError("Speeder not found; execution should have never reached here.")
```

### Motivation, pitch

For discrete-label (sort of) tasks such as speech recognition, knowing the factor upon which the speed perturbation will be applied is not so important. But for tasks where the label is segment-based, such as voice activity detection, the labels consist of boundaries in which an event occurs in time, and those annotations *change* depending on whether the signal was speeded up or slowed down.

In the latter case, adjusting the labels would be just a matter of multiplying the timestamsp by the factor.

### Alternatives

IMO a better way to handle this would be to make the [`Speed`](https://github.com/pytorch/audio/blob/main/torchaudio/transforms/_transforms.py#L1953) class itself to return the factor, simply because 1) it is already being tracked as a class variable `self.factor` at the constructor, and 2) a "flat" tuple `(x, y, z)` would be prettier to return than a "nested" one `((x, y), z)` 😋 

```python
class Speed(torch.nn.Module):

    ...

    def forward(self, waveform, lengths: Optional[torch.Tensor] = None) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:

        ...

        if lengths is None:
            out_lengths = None
        else:
            out_lengths = torch.ceil(lengths * self.target_sample_rate / self.source_sample_rate).to(lengths.dtype)

        return self.resampler(waveform), out_lengths, self.factor  # XXX <- change here
```

**Comment by cassiotbatista** [comment]
Just realized it makes more sense to divide the tensor's original shape by the new one -- it is more accurate and effortless. Closing it :)
